doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -0,0 +1,571 @@
+    "keybindings": []
+}
+
+```
+
+Note the following:
+* cmd.exe and powershell.exe are both in the file, as to give users an easy
+  point to extend the settings for those default profiles.
+* Powershell Core is included in the file, and the default profile has been set
+  to its GUID.
+* There are a few helpful comments scattered throughout the file to help point
+  the user in the right direction.
+
+### Re-ordering profiles
+
+Since there are shortcuts to open the Nth profile in the list of profiles, we
+need to expose a way for the user to change the order of the profiles. This was
+not a problem when there was only a single list of profiles, but if the defaults
+are applied _first_ to the list of profiles, then the user wouldn't be able to
+change the order of the default profiles. Additionally, any profiles they add
+would _always_ show up after the defaults.
+
+To remedy this, we could scan the user profiles in the user settings first, and
+create `Profile` objects for each of those profiles first. These `Profile`s
+would only be initialized with their GUID temporarily, but they'd be placed into
+the list of profiles in the order they appear in the user's settings. Then, we'd
+load all the default settings, overlaying any default profiles on the `Profile`
+objects that might already exist in the list of profiles. If there are any
+default profiles that don't appear in the user's settings, they'll appear
+_after_ any profiles in the user's settings. Then, we'll overlay the full user
+settings on top of the defaults.
+
+## UI/UX Design
+
+### Opening `defaults.json`
+How do we open both these files to show to the user (for the interim period
+before a proper Settings UI is created)? Currently, the "Settings" button only
+opens a single json file, `profiles.json`. We could keep that button doing the
+same thing, though we want the user to be able to also view the default settings
+file, to be able to inspect what settings they wish to change.
+
+We could have the "Settings" button open _both_ files at the same
+time. I'm not sure that `ShellExecute` (which is used to open these files)
+provides any ordering guarantees, so it's possible that the `defaults.json`
+would open in the foreground of the default json editor, while making in unclear
+that there's another file they should be opening instead. Additionally, if
+there's _no_ `.json` editor for the user, I believe the shell will attempt
+_twice_ to ask the user to select a program to open the file with, and it might
+not be clear that they need to select a program in both dialogs.
+
+Alternatively, we could make the defaults file totally inaccessible from the
+Terminal UI, and instead leave a comment in the auto-generated `profiles.json`
+like so:
+
+```json
+// To view the default settings, open the defaults.json file in this directory
+```
+
+The "Settings" button would then only open the file the user needs to edit, and

```

Comment:
I would leave the comment in the settings.json to talk about where defaults.json is AND add some sort of Alt+LeftClick detection on the Settings button that opens the defaults.json instead of settings.json (and leave a message at the top of defaults.json that says "click the button without alt to open the settings.json one" perhaps)

Reply:
done